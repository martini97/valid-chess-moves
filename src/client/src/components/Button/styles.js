export default theme => ({
  button: {
    margin: theme.spacing.unit,
    [theme.breakpoints.down('sm')]: {
      width: '90vmin',
    },
    width: '80vmin',
  },
  container: {
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
  },
  wrapper: {
    position: 'relative',
  },
});
