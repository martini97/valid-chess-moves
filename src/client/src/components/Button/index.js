import React from 'react';
import Proptypes from 'prop-types';
import {
  Button as MUIButton,
  CircularProgress,
} from '@material-ui/core';
import { withStyles } from '@material-ui/core/styles';

import styles from './styles';

const Button = ({ classes, disabled, handleClick, color, children, isLoading }) => (
  <div className={classes.container}>
    <MUIButton
      variant="contained"
      color={color}
      className={classes.button}
      disabled={disabled}
      onClick={handleClick}
    >
      {
        isLoading
        ? <CircularProgress size={20} />
        : children
      }
    </MUIButton>
  </div>
);

Button.propTypes = {
  classes: Proptypes.object.isRequired,
  children: Proptypes.node.isRequired,
  handleClick: Proptypes.func.isRequired,
  disabled: Proptypes.bool,
  color: Proptypes.string,
  isLoading: Proptypes.bool,
};

Button.defaultProps = {
  color: 'primary',
  disabled: false,
  isLoading: false,
};

export default withStyles(styles)(Button);
