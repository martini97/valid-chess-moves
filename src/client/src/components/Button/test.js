import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import ReactTestUtils from 'react-dom/test-utils';
import { render, mount } from 'enzyme';
import { Button as MUIButton } from '@material-ui/core';

import Button from '.';

it('renders without crashing', () => {
  const div = document.createElement('div');

  const props = { handleClick: () => null }
  ReactDOM.render(
    <Button {...props}>
      children
    </Button>,
    div);
  ReactDOM.unmountComponentAtNode(div);
});

it('snapshot', () => {
  const props = { handleClick: () => null }
  const component = renderer.create(
    <Button {...props}>
      children
    </Button>
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it('should contain mui button', () => {
  const props = { handleClick: () => null }
  const wrapper = mount(
    <Button {...props}>
      children
    </Button>
  );
  expect(wrapper.contains(MUIButton)).toEqual(true);
});

it('should call function on click', () => {
  const props = { handleClick: jest.fn() }
  const wrapper = mount(
    <Button {...props}>
      children
    </Button>
  );

  wrapper.find('button').simulate('click');
  expect(props.handleClick.mock.calls.length).toBe(1);
});

it('should call function on click when disabled', () => {
  const props = { handleClick: jest.fn(), disabled: true }
  const wrapper = mount(
    <Button {...props}>
      children
    </Button>
  );

  wrapper.find('button').simulate('click');
  expect(props.handleClick.mock.calls.length).toBe(0);
});
