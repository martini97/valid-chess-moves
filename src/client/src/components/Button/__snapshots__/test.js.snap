// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<div
  className="Button-container-38"
>
  <button
    className="MuiButtonBase-root-63 MuiButton-root-39 MuiButton-contained-48 MuiButton-containedPrimary-49 MuiButton-raised-51 MuiButton-raisedPrimary-52 Button-button-37"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex="0"
    type="button"
  >
    <span
      className="MuiButton-label-40"
    >
      children
    </span>
    <span
      className="MuiTouchRipple-root-66"
    />
  </button>
</div>
`;
